### DISCUSSION TITLE
Mastodon::UnexpectedResponseError ... returned code 401

### DISCUSSION THREAD

**Discussion by Epiphero** [mastodon/mastodon#20983] [Q&A]
I've just set up an instance and it kinda sorta seems to be working, except I can't seem to follow anyone on any other instances.  Based on a reddit post i tried searching for https://mastodon.social/@Gargron on my instance, and after a while it finds him, and there is a follow button.  But that seems to be as far as things go.  If I go to my sidekiq dashboard, I see that there is ActivityPub::DeliveryWorker job in the push queue with the following error:

```
Mastodon::UnexpectedResponseError: https://mastodon.social/users/Gargron/inbox returned code 401
```

This has been retrying for a while with no indication of anything other than the 401.  Where could I look for further log messages like perhaps the response that came with the 401 code?  Thanks for any help

**Comment by Epiphero** [comment]
OK, a couple of days later and one of the follows was successful, but the other one I tried is still retrying.  Is it common to take days for a follow request to go through?  Maybe a result of the great migration ... I see that I seem to be able to follow more accounts if I try fresh... I guess I'll wait another day or two and see if the hung up one follows suit. ¯\_(ツ)_/¯

**Comment by lazzarello** [comment]
I'm seeing the same thing on a server I got going 2 hours ago. Is there some kind of hidden authentication that federated servers perform? Perhaps there's a time-based block list on new follows from servers with less than x days uptime? Looking through docs for some clues.

It seems there might be a DNS timeout problem? I did change the DNS record a couple days ago.

https://github.com/mastodon/mastodon/discussions/21404

**Reply by Epiphero** [reply]
FWIW, things did eventually start going through for me.  It was mostly a mastodon.online problem too...

**Reply by alisyd** [reply]
Hey I have recently migrated my mastodon instance to a new machine, and now I am getting this error, was your error resolved with time, do I have to wait or am I missing something here?

**Comment by nfriedly** [comment]
I had this problem and I finally tracked it down to being a /.well-known/webfinger issue:

I had set LOCAL_DOMAIN to nfriedly.com and WEB_DOMAIN to mastodon.nfriedly.com so that mu profile would be @nfriedly@nfriedly.com even though the server is actually at mastodon.nfriedly.com. But I had overlooked the webfinger info at the end of https://docs.joinmastodon.org/admin/config/#web_domain - basically every other federated server was trying to communicate with nfriedly.com instead of mastodon.nfriedly.com., and then failing.

They provide an example nginx config to hande this with a 301 redirect for /.well-known/webfinger, but I had nfriedly.com hosted on github pages which doesn't allow you to do that kind of configuration.

I initially switched LOCAL_DOMAIN to mastodon.nfriedly.com, and this made my feed start getting posts from other servers. But changing it reportedly breaks other things and I still wanted my initial setup. So....

I ended up reverting that and changing my DNS so that www.nfriedly.com is still on github pages, but nfriedly.com (without any subdomain) now goes to my mastodon server, which redirects everything to www.nfriedly.com except for the /.well-known/webfinger, which gets redirected to mastodon.nfriedly.com.

Now I just have to wait a day or two for DNS to propagate to see if this will actually work...
